builders/tslice.py

```diff
@@ -0,0 +1,190 @@
+           "groupby_attrs": ["time_range"],
+           "aggregations":[
+                           {
+                           "type": "union",
+                           "attribute_name": "time_range"
+                           },
+                           {
+                           "type": "join_existing",
+                           "attribute_name": "time_range",
+                           "options": {
+                                      "dim": "time",
+                                      "coords": "minimal",
+                                      "compat": "override"
+                                      }
+                           },
+           ] 
+        }                         
+    }
+    # add a section for each column
+    for col in cols:
+        c = {
+            "column_name": col,
+            "vocabulary": ""
+            }
+        d["attributes"].append(c)
+
+    od=OrderedDict(d.items())
+    #write the json file
+    with open(json_filepath,'w') as f:
+        json.dump(od,f,indent=4)
+
+
+def verify(input_yaml):
+    # verify that we're working with a dictionary
+    if not isinstance(input_yaml,dict):
+        print("ERROR: The experiment/dataset top level is not a dictionary. Make sure you follow the correct format.")
+        return False
+    for dataset in input_yaml.keys():
+        # check to see if there is a data_sources key for each dataset
+        if 'data_sources' not in input_yaml[dataset].keys():
+            print("ERROR: Each experiment/dataset must have the key 'data_sources'. Verify "+dataset+" contains this key.")
+            return False
+        # verify that we're working with a list at this level
+        if not isinstance(input_yaml[dataset]['data_sources'],list):
+            print("ERROR: The data_sources are not in a list.  Make sure you follow the correct format.")
+            return False
+        for stream_info in input_yaml[dataset]['data_sources']:
+            # check to make sure that there's a 'glob_string' key for each data_source
+            if 'glob_string' not in stream_info.keys():
+                print("ERROR: Each data_source must contain a 'glob_string' key.  Verify that all data_sources under "+dataset+" contain a 'glob_string' key.")
+                return False
+    return True
+
+
+def common_parser(filepath, local_attrs, glob_attrs):
+    """
+    gather the file parts for the df
+    """
+    basename = os.path.basename(filepath)
+    fileparts = {}
+    fileparts['path'] = filepath
+
+    try:
+        fileparts['variable'] = []
+        # open file
+        d = nc.Dataset(filepath,'r')
+        # find what the time (unlimited) dimension is
+        dims = list(dict(d.dimensions).keys())
+        look_for_unlim=[d.dimensions[dim].isunlimited() for dim in dims]
+        unlim=[i for i, x in enumerate(look_for_unlim) if x]
+        unlim_dim=dims[unlim[0]]
+        fileparts['time_range'] = str(d[unlim_dim][0])+'-'+str(d[unlim_dim][-1])
```

Comment:
I tried running this code section using one of the WRF files: `/glade/collections/cdg/work/cordex/raw/wrf-era-25/1979/2D/wrfout_d01_1979-03-28_00:00:00`, but it fails:

<img width="1130" alt="Screen Shot 2020-04-21 at 3 08 14 PM" src="https://user-images.githubusercontent.com/13301940/79914130-f9cc0880-83e1-11ea-99b7-8cbdbeb15454.png">


Do you know what's happening?

Reply:
It would be nice to have a set of files from different models that we could use to set up a testing framework for this. I am happy to help set this up.

Reply:
I have not tested this on WRF files, but from what I remember, I think this isn't working because Time is just a coordinate and not a variable.  So I'll need another way to get the time range for WRF files.

Reply:
Thanks @andersy005 for looking at this draft.  That would be great if we can create a set of files we can test with.

Reply:
@sherimickelson When you say "Time is just a coordinate," are you saying "Time is just a dimension"?

Reply:
I ask because in `xarray` (and according to the CF conventions) a "coordinate" *is* a type of "variable."  And a "dimension" is just a name with a size.

Reply:
@kmpaul, yes, I'm sorry.  It is just a dimension.